cli/config/config.go

```diff
@@ -90,18 +90,18 @@ func Load(configDir string) (*configfile.ConfigFile, error) {
 	}
 
 	// Can't find latest config file so check for the old one
-	confFile := filepath.Join(homedir.Get(), oldConfigfile)
+	confFile := filepath.Join(configDir, oldConfigfile)
```

Comment:
Not sure this is actually correct; custom locations for the configuration file were added in https://github.com/moby/moby/pull/12856, but at that time, `.dockercfg` was already the _old_ file, and replaced with `~/.docker/config.json` (that change was made in https://github.com/moby/moby/pull/12009)


If we change this now, we're effectively adding new functionality (don't think setting `--config-dir=/foo` would read `/foo/.dockercfg` in older versions, correct? For a deprecated feature.

I think we should print a deprecation warning if we encounter an `~/.dockercfg` file, and remove the logic to look for this filename in the next release (or the one after); this needs to be added to the deprecated features document; https://github.com/docker/cli/blob/master/docs/deprecated.md

Reply:
I do agree with @thaJeztah on this. This part of the code is looking for older/deprecated configuration file that was _hardcoded_ in `$HOME/.dockercfg`… So we shouldn't change that (otherwise we're effectively adding a new functionnality).